coordinator/src/main/scala/filodb.coordinator/queryengine2/RoutingPlanner.scala

```diff
@@ -139,10 +139,19 @@ object QueryRoutingPlanner extends RoutingPlanner {
     * Check whether logical plan has a PeriodicSeriesPlan
     */
   def isPeriodicSeriesPlan(logicalPlan: LogicalPlan): Boolean = {
-    if (logicalPlan.isInstanceOf[RawSeriesPlan] || logicalPlan.isInstanceOf[MetadataQueryPlan]) {
+    if (!logicalPlan.isRoutable) {
       false
-    } else
-    true
+    } else if (logicalPlan.isInstanceOf[ScalarVectorBinaryOperation]) {
```

Comment:
It occurs to me that this entire block would be much more readable as a match {.... }:

    logicalPlan match {
      case p: VectorPlan =>  isPeriodicSeriesPlan(p.scalars)
      case p: ApplyInstantFunction => isPeriodicSeriesPlan(p.vectors)
      ...
    }

See, no need for as/isInstanceOf